cogs/displayinfo.py

```diff
@@ -33,85 +32,139 @@ class DisplayInfo(commands.Cog):
+                if before_string_id in self.timers:
+                    if self.current_date in self.timers[before_string_id]:
+                        if mem_before.activity and mem_before.activity.name in self.timers[before_string_id][self.current_date]:
+                            currentStartTime = self.timers[before_string_id][self.current_date][mem_before.activity.name]["startTime"]
+                            self.timers[before_string_id][self.current_date][mem_before.activity.name]["startTime"] = None
+                            if currentStartTime:
+                                playTime = datetime.now() - currentStartTime # Calculate the time difference (playTime) between the start and end time of the activity
+                                self.timers[before_string_id][self.current_date][mem_before.activity.name]["totalTime"] = self.timers[before_string_id][self.current_date][mem_before.activity.name].get("totalTime", 0) + playTime.total_seconds() # Add the play time to the member's total play time for that activity for that day
+                            print("end:", self.timers) # printing the dictionary to see the updates after an activity is ended
+                            self.current_collection.replace_one({}, self.timers) # Update the MongoDB collection with the updated activity data(self.timers)
+
+    #This function periodically updates the activities of members that are online and in an activity.
+    async def timed_update(self):
+        while len(self.members_online) > 0:
             await asyncio.sleep(300)
+            for memberID, guildID in self.members_online.items():
+                guild = self.bot.get_guild(guildID)
+                member = guild.get_member(memberID)
+                update_success = await self._update_now(member)
+            
+            print("timed update:", self.timers)
+            self.current_collection.replace_one({}, self.timers)
+
+    #This function resets all the startTimes of a member
+    async def reset_start(self, member):
+        string_id = str(member.id)
+        if string_id in self.timers and self.current_date in self.timers[string_id]:
+            for activity, info in self.timers[string_id][self.current_date].items():
+                info["startTime"] = None
             
 
     # This is a Discord bot command that retrieves and displays activity time information for a given member.
     @commands.command()
-    async def get_info(self, member):
-        if member.author.name in self.timers:
-            channel = member.channel
+    async def get_info(self, ctx):
+        guild = self.bot.get_guild(ctx.guild.id)
+        string_id = str(ctx.author.id)
+        member = guild.get_member(ctx.author.id)
+
+        update_success = await self._update_now(member)
+        print("get info update:", self.timers)
+        self.current_collection.replace_one({}, self.timers) 
+        if string_id in self.timers:
```

Comment:
New Simple Ticket (Maybe something to assign to @shakeelriyaj): For future purposes, we can try to return a simple message to those that don't have any information in the database but decide to call `get_info()`.